metpy/calc/indices.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2008-2015 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Contains calculation of various derived indicies."""
+import numpy as np
+
+from . import mixing_ratio, saturation_vapor_pressure
+from .tools import get_layer
+from ..constants import g, rho_l
+from ..package_tools import Exporter
+from ..units import units
+
+exporter = Exporter(globals())
+
+
+def precipitable_water(dewpt, p, top=400 * units('hPa')):
+    r"""Calculate precipitable water through the depth of a sounding.
+
+    Default layer depth is sfc-400 hPa. Formula used is:
+
+    \begin{align}
+    \frac{1}{pg} \int\limits_0^d x \,dp
+    \end{align}
+
+    from [Tsonis2008]_, p. 170.
+
+    Parameters
+    ----------
+    dewpt : array-like
+        Atmospheric dewpoint profile
+    p : array-like
+        Atmospheric pressure profile
+    top: `pint.Quantity`
+        The top of the layer, specified in pressure.
+
+    Returns
+    -------
+    `pint.Quantity'
+        The precipitable water in the layer, in inches
+
+    """
+    pw_layer, dewpt_layer = get_layer(p, dewpt, depth=p[0] - top)
+
+    w = mixing_ratio(saturation_vapor_pressure(dewpt_layer), pw_layer)
+
+    sort_inds = np.argsort(pw_layer)
+    pw_layer = pw_layer[sort_inds]
+    w = w[sort_inds]
+
+    pw = np.trapz(w, pw_layer) / (g * rho_l)
+    pw = ((pw.magnitude * 100) * units('meter')).to('inches')
```

Comment:
I'd vote for mm as it would be good to at least stick with metric units as the default behavior. To make the code a bit clearer, I think I might do the unit attachment on the return/separate step so that one liner is less dense.